### DISCUSSION TITLE
use RAUC streaming and adaptive updates

### DISCUSSION THREAD

**Discussion by jluebbe** [home-assistant/operating-system#2365] [Ideas]
As one of the RAUC maintainers, I was looking around for examples to mention in our documentation and [noticed](https://github.com/home-assistant/operating-system/blob/dev/buildroot-external/ota/manifest.raucm.gtpl) that this project still uses the old bundle format ("plain").  The replacement format ("verity") was introduced in [1.5](https://rauc.readthedocs.io/en/latest/changes.html#release-1-5-released-dec-14-2020) and by now supports some features that might be useful ([HTTPS streaming](https://rauc.readthedocs.io/en/latest/advanced.html#http-streaming) and [adaptive updates](https://rauc.readthedocs.io/en/latest/advanced.html#sec-adaptive-updates)).

RAUC 1.8 is in the current buildroot release, so the main consideration is if old installations are not setup correctly to use verity bundles yet (mainly it requires RAUC 1.5 and dm-verity enabled in the kernel). To support those, a transition period could be used, where support for updating from verity bundles is enabled, but plain bundles are still generated for updates.

In the next release, you could offer verity bundles with streaming and adaptive update. Additionally a plain bundle could be offered for direct updates from older versions.

We'd be happy to help with the details of the transition, if needed.

**Comment by agners** [comment]
As we discussed elsewhere, I've added verity a couple of weeks back (see https://github.com/home-assistant/operating-system/pull/2622).

I've tested upgrade with a rauc bundle with the verity format, and it seems to just work :+1: 

The Supervisor controls OS updates, and it makes sure that we only update from major to the next major. We also make sure we update to the last previous major. Currently that would be 10.5 with all the kernel modules for the verity format in already. So with Home Assistant 11 we can switch to verity format. I plan to do some more testing, but I intend to do the switch with OS 11.

**Reply by agners** [reply]
FWIW, currently, I do not intend to use HTTPS streaming or adaptive updates. The update files (typically 150MB) are relatively small (compared to Home Assistant Core itself). So I don't really see the value introducing these features for us right now.

That said, verity seems to be the more future proof format, and it doesn't really have downsides. Installation seems to be slightly faster as well. So I don't really see a reason not to update.
